src/pages/product/[id].tsx

```diff
@@ -30,40 +30,43 @@ export default function ProductDetail({
   product,
   error,
 }: {
-  product: Product;
-  error: { message: string };
+  product?: Product;
+  error?: { message: string };
 }) {
-  const [totalPrice, setTotalPrice] = useState<number>(product.price || 0);
+  const [totalPrice, setTotalPrice] = useState<number>(product?.price || 0);
 
   const handlePriceChange = (e: ChangeEvent<HTMLInputElement>) => {
-    const newTotalPrice = calculateTotalPrice(e, product.price);
+    const newTotalPrice = calculateTotalPrice(e, product?.price);
     setTotalPrice(newTotalPrice);
   };
-
   if (error) {
```

Comment:
```suggestion
if (error || !product) {
```
Suggest: 이렇게하면, 아래에서 product를 optional하게 처리안해도 될 것 같습니다.

Reply:
이런 방법을 생각 못했네요..! 감사합니다 [6203088](https://github.com/house-study/guddnboy-market/pull/5/commits/6203088933422247b58fa8d1cd4395c5f3ea373c)